opm/grid/cpgrid/CpGrid.cpp

```diff
@@ -1491,10 +1491,15 @@ void CpGrid::selectWinnerPointIds([[maybe_unused]] std::vector<std::vector<int>>
 void CpGrid::populateCellIndexSetRefinedGrid([[maybe_unused]] int level)
 {
 #if HAVE_MPI
-    const auto& level_global_id_set =  (*current_data_)[level]->global_id_set_;
-    auto& level_index_set =  currentData()[level]->cellIndexSet();
+    const auto& level_data = currentData()[level];
+    const auto& level_global_id_set =  level_data->global_id_set_;
+    auto& level_index_set =  level_data->cellIndexSet();
 
-    level_index_set.beginResize();
+    // Redefine the level cell index set by creating a new ParallelIndexSet 
+    // and then swapping it with the existing one.
+    ParallelIndexSet new_level_index_set; 
+
+    new_level_index_set.beginResize();
```

Comment:
Too bad that ParallelIndexSet is such an old fashioned and incomplete container (no clear method, no swap, no move ...). Hence I think swapping at the end of this function is probably more costly than
```
level_index_set = ParallelIndexSet();
level_index_set.beginResize();
```
without creating the additional new_level_indexset.

I might be missing something here and maybe you already tried that?

Reply:
Wait a minute. Now I am sceptical about my own hypothesis..

Reply:
So I need to correct myself. As we only use std containers or custome containers that use standard containers and do not implement copy or assignment. The containers should even have autogenerated move constructors and swap should be efficient.

We still need extra space because of two containers.